### PR TITLE
Restrict timesheet entries to be added to only one invoice

### PR DESCRIPTION
**Pull request by mohinid**
## Notion card
https://www.notion.so/saeloun/Bug-If-a-timesheet-entry-has-been-added-to-an-invoice-in-draft-status-and-if-I-create-new-invoice--6255fd6ba598482eaa30e0363afcd093

## Summary
If a timesheet entry was added to an invoice in draft status and if we create a new invoice then those entries were still available to be added to the new invoice. This pr fixes this bug and restrict entries to be displayed only if they are not a part of any other invoice.

## Type of change

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
On local

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

File: app/controllers/internal_api/v1/generate_invoice_controller.rb [hunk 1]
```diff
@@ -32,8 +32,15 @@ def search_term
       @search_term ||= (params[:search_term].present?) ? params[:search_term] : "*"
     end
 
+    # merging selected_entries from params with ids already present in other invoice line items
+    def filtered_ids
+      invoice_line_item_ts_ids = InvoiceLineItem.pluck(:timesheet_entry_id).uniq
```
Comment (akhilgkrishnan) [inline app/controllers/internal_api/v1/generate_invoice_controller.rb:37]:
```suggestion
      invoice_line_item_ts_ids = InvoiceLineItem.joins(:timesheet_entry).where(:timesheet_entry => {bill_status: "unbilled"}).pluck(:timesheet_entry_id).uniq
```

**Comment by mohinid** [comment]
Add test for change.

**Review by apoorv1316 — APPROVED** [review]

**Comment by mohinid** [comment]
> TODO: Add test for change.

@akhilgkrishnan  please review and approve.

**Comment by supriya3105** [comment]
@prasanthchaduvula Can you review this PR ?

**Review by akhilgkrishnan — APPROVED** [review]
LGTM

**Pull request merged**